packages/spectral/src/types/ConfigVars.ts

```diff
@@ -0,0 +1,327 @@
+   * "embedded" - Customers cannot view or update the Config Var as the value will be set programmatically.
+   * "organization" - Customers cannot view or update the Config Var as it will always have a default value or be set by the organization.
+   */
+  permissionAndVisibilityType?: PermissionAndVisibilityType;
+  /** Optional value that specifies whether this Config Var is visible to an Organization deployer. @default true */
+  visibleToOrgDeployer?: boolean;
+};
+
+type GetDynamicProperties<
+  TValue,
+  TCollectionType extends CollectionType | undefined
+> = TCollectionType extends undefined
+  ? {
+      defaultValue?: TValue;
+      /** Optional value to specify the type of collection if the Config Var is multi-value. */
+      collectionType?: undefined;
+    }
+  : TCollectionType extends "valuelist"
+  ? {
+      defaultValue?: TValue[];
+      /** Optional value to specify the type of collection if the Config Var is multi-value. */
+      collectionType: "valuelist";
+    }
+  : {
+      defaultValue?:
+        | Record<string, TValue>
+        | Array<{ key: string; value: TValue }>;
+      /** Optional value to specify the type of collection if the Config Var is multi-value. */
+      collectionType: "keyvaluelist";
+    };
+
+type StandardConfigVarDynamicProperties<TDataType extends ConfigVarDataType> =
+  | CollectionType
+  | undefined extends infer TCollectionType
+  ? TCollectionType extends CollectionType | undefined
+    ? TDataType extends ConfigVarSingleDataType
+      ? TCollectionType extends undefined
+        ? GetDynamicProperties<
+            ConfigVarDataTypeDefaultValueMap[TDataType],
+            undefined
+          >
+        : never
+      : GetDynamicProperties<
+          ConfigVarDataTypeDefaultValueMap[TDataType],
+          TCollectionType
+        >
+    : never
+  : never;
+
+type CreateStandardConfigVar<TDataType extends ConfigVarDataType> =
+  BaseConfigVar &
+    StandardConfigVarDynamicProperties<TDataType> & {
+      /** The data type of the Config Var. */
+      dataType: TDataType;
+    };
+
+type StringConfigVar = CreateStandardConfigVar<"string">;
+
+type DateConfigVar = CreateStandardConfigVar<"date">;
+
+type TimestampConfigVar = CreateStandardConfigVar<"timestamp">;
+
+type PicklistConfigVar = CreateStandardConfigVar<"picklist"> & {
+  /** Optional list of picklist values. */
+  pickList: string[];
+};
+
+/** Choices of programming languages that may be used for Config Var code values. */
+export type CodeLanguageType = "json" | "xml" | "html";
+
+type CodeConfigVar = CreateStandardConfigVar<"code"> & {
+  /** Optional value to specify the type of language of a code Config Var. */
+  codeLanguage?: CodeLanguageType;
+};
+
+type BooleanConfigVar = CreateStandardConfigVar<"boolean">;
+
+type NumberConfigVar = CreateStandardConfigVar<"number">;
+
+type ScheduleConfigVar = CreateStandardConfigVar<"schedule"> & {
+  /** Optional timezone for the schedule. */
+  timeZone?: string;
```

Comment:
```suggestion
  /** Timezone for the schedule. */
  timeZone: string;
```
Same here, optional?